src/gtfs/mod.rs

```diff
@@ -355,21 +351,106 @@ pub fn read_from_zip<P: AsRef<Path>>(p: P, configuration: Configuration) -> Resu
+pub fn read<P: AsRef<Path>>(p: P) -> Result<Model> {
+    Reader::default().parse(p)
+}
+
+/// Structure to configure the GTFS reading
+#[derive(Default)]
+pub struct Reader {
+    configuration: Configuration,
+}
+
+impl Reader {
+    /// Build a Reader with a custom configuration
+    pub fn new(configuration: Configuration) -> Self {
+        Self { configuration }
+    }
+
+    /// Imports a `Model` from the
+    /// [GTFS](https://gtfs.org/reference/static).
+    /// files in the given directory.
+    /// This method will try to detect if the input is a zipped archive or not.
+    /// If the default file type mechanism is not enough, you can use
+    /// [Reader::parse_zip] or [Reader::parse_dir].
+    pub fn parse(self, path: impl AsRef<Path>) -> Result<Model> {
+        let p = path.as_ref();
+        if p.is_file() {
+            // if it's a file, we consider it to be a zip (and an error will be returned if it is not)
+            Ok(self
+                .parse_zip(p)
+                .with_context(|_| format!("impossible to read zipped gtfs {:?}", p))?)
+        } else if p.is_dir() {
+            Ok(self
+                .parse_dir(p)
+                .with_context(|_| format!("impossible to read gtfs directory from {:?}", p))?)
+        } else {
+            Err(failure::format_err!(
+                "file {:?} is neither a file nor a directory, cannot read a gtfs from it",
+                p
+            ))
+        }
+    }
+
+    /// Imports a `Model` from a zip file containing the
+    /// [GTFS](https://gtfs.org/reference/static).
+    pub fn parse_zip(self, path: impl AsRef<Path>) -> Result<Model> {
+        let reader = std::fs::File::open(path.as_ref())?;
+        let mut file_handler = read_utils::ZipHandler::new(reader, path)?;
+        read_file_handler(&mut file_handler, self.configuration)
+    }
+
+    /// Imports a `Model` from the [GTFS](https://gtfs.org/reference/static)
+    /// files in the `path` directory.
+    pub fn parse_dir(self, path: impl AsRef<Path>) -> Result<Model> {
+        let mut file_handler = read_utils::PathFileHandler::new(path.as_ref().to_path_buf());
+        read_file_handler(&mut file_handler, self.configuration)
+    }
+
+    /// Imports a `Model` from an object implementing `Read` and `Seek` and containing the
+    /// [GTFS](https://gtfs.org/reference/static).
+    ///
+    /// This method makes it possible to read from a variety of sources like read a GTFS
+    /// from the network.
+    ///
+    /// ```ignore
+    /// let url = "http://some_url/gtfs.zip";
+    /// let resp = reqwest::blocking::get(url)?; // or async call
+    /// let data = std::io::Cursor::new(resp.bytes()?.to_vec());
+    /// let model = transit_model::gtfs::Reader::default().parse_zip_reader(data, &url)?;
+    /// # Ok::<(), Error>(())
+    /// ```
+    ///
+    /// The `source_name` is needed to have nicer error messages.
+    pub fn parse_zip_reader<R>(self, reader: R, source_name: &str) -> Result<Model>
```

Comment:
On naming (if not already discussed, too): It looks like a `bulk_reader` (or `file_set_reader`), not especially a zip.

Reply:
yes, we decided to add `zip` in the name, because the reader should be a `zip` ([see here](https://github.com/CanalTP/transit_model/pull/780/files#diff-249da72fb614dd947b88beb2e4d96fa07412868a711f75506e0b2fd3ed9746ccR451))